packages/formation-react/src/components/AccessibleLink/AccessibleLink.mdx

```diff
@@ -0,0 +1,113 @@
+---
+title: AccessibleLink
+name: AccessibleLink
+tags: accessible, accessiblelink, component
+---
+
+import AccessibleLink, {
+  CONTACTS,
+  PATTERNS,
+  MEDIA_SITES,
+  FILE_TYPES,
+  FILE_SIZES,
+} from './AccessibleLink';
+
+### Code:
+
+```javascript
+import AccessibleLink, {
+  CONTACTS,    // Telephone contacts
+  PATTERNS,    // Telephone pattern
+  MEDIA_SITES, // MediaLink sites
+  FILE_TYPES,  // DownloadLink known MIME-types
+  FILE_SIZES,  // DownloadLink known file sizes (B, KB, MB & GB)
+} from '@department-of-veterans-affairs/formation-react/AccessibleLink';
+
+// Download minimal example
+<AccessibleLink.Download
```

Comment:
This component's interface is pretty focused on the use case of a static PDF/file. Another common use case is a dynamically generated file or a file based on structured api data. That's mostly what I've had to deal with when building file links on va.gov and this interface doesn't work all that well for those cases, which would definitely benefit from the a11y work in here. Is it worth gathering some information on that use case before making this a formation component?

Reply:
I tried to make this component flexible enough to work in most scenarios. Would you please share an example of how it would be difficult to use this component? I guess we can continue this discussion in slack, maybe in the #vetsgov-accessibility channel?

Reply:
@jbalboni Yes, this is something I'm concerned about. The recent BAM2 Debt Letters use case and the download attribute issue with the Content-Disposition HTTP header's filename parameter — frankly some of it I don't yet understand completely — gave me pause about including the download attribute without some further exploration.  Do y'all or @rianfowler or @1Copenut have ideas on how we can support teams with this? Should we use this attribute? If yes, can we provide guidance for how to use it and the content-disposition header parameters ([review Slack thread](https://dsva.slack.com/archives/CBU0KDSB1/p1596751922192300?thread_ts=1596742928.177700&cid=CBU0KDSB1))?

Reply:
I'm not sure about the use case with the `download` attribute and a file from api. On VAOS we have a generated ICS file that uses the `download` attribute and handles IE separately: https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/vaos/components/AddToCalendar.jsx#L27. It might be taking on too much to incorporate that kind of logic in this component, I'm not really sure.

Reply:
I've updated the components to now include a `content` callback function to render any custom link content. I've also updated the demo - https://plnkr.co/edit/aOhgB5ZjPx8IJb53; please let me know what you think.

Reply:
I think the `A link to Download a huge legacy MS document file DOC (1.1GB).` example shows what I'm worried about. It didn't download. When I tested the BAM2 download letter pdf the browser hung. Maybe plnkr just shows the unhappy face instead?

Reply:
> It didn't download

That example isn't using a real URL - https://example.va.gov/big-doc.doc - so I wouldn't expect it to work. The "Download this random json (blob) file" however will work; it's a "created" download link that includes content I created within that demo.

Reply:
That link starts with `blob`, though, which doesn't replicate the issue with the BAM2 debt letters scenario that doesn't respect the download attribute, also seems to hang unless opened in a new tab. Would it be easier if we chatted about this on a call on Wednesday?

Reply:
Yes, I would appreciate it.

Reply:
From our meeting last week regarding a download link using a dynamically generated file or a file based on structured api data, it was decided:

- The file name would be established by the backend dev in the Content-Disposition HTTP header's filename parameter using meta data, working with the designer and accessibility to ensure a meaningful filename.
- The `download` attribute would not be included in the link.
- It may be helpful to add target="_blank" to the download link if there is a delay in downloading the file, as it signals to the user that something is happening, as long as the new tab/window closes once the download is complete.

Reply:
@jenstrickland The `download` attribute has 2 settings:
* `download` (equivalent to `download="true"` sets the link as a downloadable file; this is necessary to establish the link behavior
* `download="filename.ext"` sets the filename and extension of the downloaded file; but, as we discovered, not all browsers will use this setting.
* Currently, this component doesn't set these download links `target="_blank"` attribute, but it would if you passed in an `external` (`true`) property. Maybe I could make it `true` by default then?

Reply:
We don't want a `target="_blank"` by default, only available as needed for cases such as the BAM2 generated file that seemed unresponsive while it generated the file for download.

Reply:
DOH, I just changed, and reverted, making download links use `target="_blank"` by default.